sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -254,52 +238,44 @@ private Transforms() {
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key,  fieldTime, valueText, FieldValueType.TIME);
                 break;
             case DATE:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.DATE)
-                    .setFormFieldDate(fieldValue.getValueDate()), valueText);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key,  fieldValue.getValueDate(), valueText, FieldValueType.DATE);
                 break;
             case INTEGER:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.INTEGER)
-                    .setFormFieldInteger(fieldValue.getValueInteger()), valueText);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key, fieldValue.getValueInteger(), valueText, FieldValueType.INTEGER);
                 break;
             case NUMBER:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.FLOAT)
-                    .setFormFieldFloat(fieldValue.getValueNumber()), valueText);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key, fieldValue.getValueNumber(), valueText, FieldValueType.FLOAT);
                 break;
             case ARRAY:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), null, key,
-                    new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.LIST)
-                        .setFormFieldList(toFormFieldArray(fieldValue.getValueArray(), readResults)), null);
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), null, key,
+                     toFormFieldArray(fieldValue.getValueArray(), readResults), null, FieldValueType.LIST);
                 break;
             case OBJECT:
-                value = new FormField(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
-                    key, new com.azure.ai.formrecognizer.models.FieldValue(FieldValueType.MAP)
-                    .setFormFieldMap(toFormFieldObject(fieldValue.getValueObject(), pageNumber, readResults)), valueText
-                );
+                value = new FormField<>(setDefaultConfidenceValue(fieldValue.getConfidence()), labelText,
+                    key, toFormFieldObject(fieldValue.getValueObject(), pageNumber, readResults), valueText,
+                    FieldValueType.MAP);
```

Comment:
Is this expected? An object is mapped to a `FieldValuteType.MAP`?

Reply:
Yes, this is mapped correctly. So for a `FieldValueType.OBJECT` the service returns a it as a Map<String. FieldValue>. So rather than calling this enum Object we expose it as a Map to the user.
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/models/FieldValue.java#L262